### ISSUE TITLE
Function calling (tools) support

### ISSUE THREAD

**Issue by lmbelo** [lmstudio-ai/lmstudio-bug-tracker#26]
Plans for function calling support?

**Comment by yagil** [comment]
Yes. Will update here when there's a beta available

**Comment by bytecod3r** [comment]
This would be awesome feature to have.

**Comment by Trippnology** [comment]
I would love to have this too. Most models that support function calling seem to implement it as a role:

```JS
{ role: 'tool', content: 'A response from a tool call' }
```

Due to the type rules, this currently causes an error, so I've been passing the tool responses back as `user` messages, which often works, but is less than ideal.

**Comment by eylvisaker** [comment]
Yes I would love this. I extended [elbruno's example for connecting Semantic Kernel to LM Studio with the metadata for tools](https://github.com/eylvisaker/semantickernel-localLLMs), but LM Studio doesn't seem to pay attention to the tools and tool_choice parameters.

**Comment by yagil** [comment]
We're about to start a beta for this. If you're interested, please fill out this google form: https://docs.google.com/forms/d/e/1FAIpQLSfqpRunKgv2ui4_CWyeQ88gFtOG6CEqFSbpRrHWgJjs2mIodw/viewform?usp=sf_link

**Comment by yagil** [comment]
Landed in 0.3.6 a few months ago https://lmstudio.ai/blog/lmstudio-v0.3.6
